### PR TITLE
ELPP-3732 Correct redirects host

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 1]
```diff
@@ -57,17 +57,17 @@ server {
     root /srv/journal/web;
 
     {% if pillar.elife.domain == 'elifesciences.org' %}
```
Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: Would be useful to cover the old-style article URIs in end2end.

**Comment by giorgiosironi** [comment]
Context for posterity: this should come from the fact that Fastly rewrites `Host` to the origin hostname, adding a separate `X-Forwarded-Host` with the original value.

File: salt/journal/config/etc-nginx-sites-available-journal.conf (hunk 1 above)
Comment (giorgiosironi) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: This `if` is misleading, `pillar.elife.domain` is always `elifesciences.org` in all environments. So this configuration would be available there (although not redirecting to `https://end2end--cdn-journal.elifesciences.org/`)

Comment (thewilkybarkid) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Ah.

We want to be using the same values as https://github.com/elifesciences/journal-formula/blob/e23915585f969fd9530e7a08e57814b829bf3e71/salt/journal/config/srv-journal-app-config-parameters.yml#L43-L57

Comment (giorgiosironi) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: `pillar.journal.default_host` is the right call, although it is mostly `null` hence this fallback. Should be populated for known environments in `builder-private`.

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 2]
```diff
@@ -1,8 +1,14 @@
-{# todo: mass redirects are useful in general. lets find a way of supporting this that isn't elife-specific #}
-{% if pillar.elife.domain == 'elifesciences.org' %}
-    include /etc/nginx/traits.d/redirect-existing-paths.conf;
+{% if pillar.journal.get('default_host') %}
+    {% set public_host = pillar.journal.get('default_host') %}
+{% elif salt['elife.only_on_aws']() %}
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: this should be backward compatible as `end2end` where `default_host` isn't defined yet would fall back to `end2end--journal.elifesciences.org`, which is the non-CDN hostname but better than nothing

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 3]
```diff
@@ -84,7 +88,7 @@ server {
     }
     {% endif %}
 
-    {% if pillar.elife.env == 'prod' and pillar.elife.domain == 'elifesciences.org' %}
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: As I suspected, these checks were coming from `thedailybugle.org` so they can go now https://github.com/elifesciences/journal-formula/pull/17

**Pull request merged**